mobile/mysterium/openvpn_connection_setup.go

```diff
@@ -128,24 +118,48 @@ func (c *openvpnConnection) Statistics() <-chan consumer.SessionStatistics {
 }
 
 func (c *openvpnConnection) Start(options connection.ConnectOptions) error {
-	newSession, clientConfig, err := c.createSession(options)
+	sessionConfig := &openvpn.VPNConfig{}
+	err := json.Unmarshal(options.SessionConfig, sessionConfig)
 	if err != nil {
 		return err
 	}
 
-	log.Info().Msgf("Client config after session create: %v", clientConfig)
-	if clientConfig.LocalPort > 0 {
-		err := c.natPinger.PingProvider(
-			clientConfig.VpnConfig.OriginalRemoteIP,
-			clientConfig.VpnConfig.OriginalRemotePort,
-			clientConfig.LocalPort,
-			clientConfig.LocalPort+1,
-			c.pingerStop,
+	if sessionConfig.LocalPort == 0 && len(sessionConfig.Ports) > 0 {
+		if len(sessionConfig.Ports) == 0 || len(c.ports) == 0 {
+			c.ports = []int{sessionConfig.LocalPort}
+			sessionConfig.Ports = []int{sessionConfig.RemotePort}
+		}
+
+		conn, err := c.natPinger.PingProvider(
+			sessionConfig.RemoteIP,
+			c.ports,
+			sessionConfig.Ports,
+			sessionConfig.LocalPort,
 		)
 		if err != nil {
 			return err
 		}
+
+		_, lPort, err := net.SplitHostPort(conn.LocalAddr().String())
```

Comment:
I believe you can get port of net.Addr without parsing string. And when no need for conversion from string to int.
```go
if addr, ok := conn.LocalAddr().(*net.UDPAddr); ok {
     addr.Port
}
```